server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestReloadSecureSettingsAction.java

```diff
@@ -68,7 +72,8 @@ public RestResponse buildResponse(NodesReloadSecureSettingsResponse response, XC
                         builder.field("cluster_name", response.getClusterName().value());
                         response.toXContent(builder, channel.request());
                         builder.endObject();
-
+                        // clear password for the original request
+                        nodesRequest.secureSettingsPassword().close();
```

Comment:
2/2 (2): The password is also cleared on the original `NodesRequest`.
The way everything fits together is as follows: The `NodeSRequest` has the password which has been read from the REST body. The password is stored as a `SecureString`. It is passed as a reference to all the constructed `NodeRequest`s. 
```
    public static class NodeRequest extends BaseNodeRequest {

        NodesReloadSecureSettingsRequest request;

        public NodeRequest() {
        }

        NodeRequest(String nodeId, NodesReloadSecureSettingsRequest request) {
            super(nodeId);
            this.request = request;
        }

        @Override
        public void readFrom(StreamInput in) throws IOException {
            super.readFrom(in);
            request = new NodesReloadSecureSettingsRequest();
            request.readFrom(in);
        }

        @Override
        public void writeTo(StreamOutput out) throws IOException {
            super.writeTo(out);
            request.writeTo(out);
        }
    }
```
ALL these requests are serialized (including the one for the local node, doing the broadcast) and their password is not cleared. The deserialized `NodeRequest`s will have their password cleared: https://github.com/elastic/elasticsearch/pull/31261/files#r195506031 .
The original password (read from the REST body) which was passed as a reference to all the broadcasted requests, which have been serialized, is cleared here.

All goes down in flames if the password in the `NodeRequest` for the local node is not copied (through serialize-deserialize) AND it is cleared before other requests are serialized.
Since ALL `NodeRequest`s are serialized this should not happen, right?

Reply:
I believe everything you've said is correct